pkg/apis/nmstate/v1/state_test.go

```diff
@@ -0,0 +1,103 @@
+package v1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	yaml "github.com/ghodss/yaml"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	var (
+		obtainedStruct   NodeNetworkState
+		obtainedManifest []byte
+		desiredState     = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: up`)
+
+		currentState = State(`
+interfaces:
+  - name: eth1
+    type: ethernet
+    state: down`)
+
+		NNSManifest = `
+apiVersion: nmstate.io/v1
+kind: NodeNetworkState
+metadata:
+  name: node01
+  creationTimestamp: "1970-01-01T00:00:00Z"
+spec:
+  managed: true
+  nodeName: node01
+  desiredState:
+    interfaces:
+      - name: eth1
+        type: ethernet
+        state: up
+currentState:
+  interfaces:
+    - name: eth1
+      type: ethernet
+      state: down
+
+`
+		NNSStruct = NodeNetworkState{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "nmstate.io/v1",
+				Kind:       "NodeNetworkState",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "node01",
+				CreationTimestamp: metav1.Unix(0, 0),
```

Comment:
we don't need CreationTimestamp here

Reply:
We do, if we don't add it it add a default value the string counterpart depends on also adding this default value, with this we just fix the value.

Maybe we can use ginkgo/gstruct to match partially the struct, but I don't really like how it shows structs when it fails.